app.js

```diff
@@ -3,129 +3,185 @@
+
+  //the while loop runs so long as both answer is not 20 and guessing is true.
+  while (answer !== 20 && guessing === true) {
+
+    //When guesses is 0 the user is alerted that the game is over.
+    if (guesses === 0) {
+      alert('You ran out of guesses!');
+      guessing = false;
+      console.log('user ran out of guesses.');
+    }
+
+    //the user is prompted for a number. That string is converted to an integer
+    //using JSON.parse().
+    var answer = parseInt(prompt('How many years have I been playing Pokemon? You have ' + guesses + ' guesses left.'));
+    console.log(answer);
+
+    //answer = JSON.parse(answer);
+    console.log('User\'s guess is ' + answer + '.');
+
+    //should the user guess correctly the loop ends and a correct answer is
+    //tallied.
+    if (answer === 20) {
+      alert('Wow, you guessed right! How did you know?');
+      guessing = false;
+      correct++;
+      console.log('comparison answer=20 happens');
+      }
+
+    //when the user's guess is too high guesses is reduced by 1 and they recieve
+    //an alert.
+    if (answer > 20) {
+      alert('too high, guess again!');
+      guesses--;
+      console.log('comparison answer>20 happens');
+      console.log('User has ' + guesses + ' guesses left.');
+    }
+
+    //when the user's guess is too low guesses is reduced by 1 and they recieve
+    //an alert.
+    if (answer < 20) {
+      alert('too low, guess again!');
+      guesses--;
+      console.log('User has ' + guesses + ' guesses left.');
+      console.log('comparison answer<20 happens');
+    }
   }
+}
 
-  //when the user's guess is too low guesses is reduced by 1 and they recieve
-  //an alert.
-  if (answer < 20) {
-    alert('too low, guess again!');
-    guesses--;
-    console.log('comparison answer<20 happens');
+function funct7() {
+  //Name that pokemon!
+  var guesses = 4;
+
+  var pokemon = ['golduck', 'honedge', 'doublade', 'aegislash'];
+  pokemon.push('greninja', 'ninjask', 'shedinja', 'celesteela');
+  pokemon.push('mew', 'magnemite', 'magnezone', 'magneton', 'drowsee');
+  console.log(pokemon);
+  while (guesses > 0) {
+    var answer = prompt('Can you guess one of my favorite Pokemon?').toLowerCase();
+
+    //answer = answer.toLowerCase();
+    if (pokemon.indexOf(answer) === -1) {
+      guesses--;
+      console.log(guesses + ': user guess');
+      console.log(pokemon.indexOf(answer) + ': index');
+      console.log(answer + ': user answer');
+      alert('Nope! Try again. You have ' + guesses + ' guesses left.');
+    } else {
+      console.log('User guessed correctly!');
+      console.log(answer + ': user answer');
+      correct++;
+      alert('Wow! You\'re a Pokemon Master!');
+      guesses = 0;
+    }
   }
 }
 
-//Name that pokemon!
-var guesses =
-var pokemon;
-pokemon = ['golduck', 'honedge', 'doublade', 'aegislash', 'greninja', 'ninjask', 'shedinja', 'celesteela', 'mew', 'magnemite', 'magnezone', 'magneton', 'drowsee'];
+funct();
```

Comment:
I would find better names for these functions, it will be hard to tell what's going on here later on.